### DISCUSSION TITLE
Multiple dynamic Domains

### DISCUSSION THREAD

**Discussion by i-am-henri** [postalserver/postal#3142] [Help with using Postal]
Hey, I have a question about domains. Can I also use “dynamic domains”? So, people can create a mx record with their DNS provider which points to my mail server? Or do I have to configure something special? Because when I try this, all I get back is “message bounced”. Or can I create new adress routes with the api?
![{B3ED8DC2-EE03-4CE4-A599-0C187799B762}](https://github.com/user-attachments/assets/515b70fa-ca75-4366-9bf9-c189a5522d01)

**Comment by willpower232** [comment]
Postal is intended to be "white-labelled" so you have one set of domains pointing at your Postal server, and listed in the configuration file, and then you can add actual domains to send with in the Postal web UI and CNAME them to the records you put in the configuration file.

If you cannot see the message status in the Postal web UI then it probably did not get there and you would have to look in some logs closer to where you are sending the message.

**Reply by i-am-henri** [reply]
So, it‘s not possible to just create an mx record and point to the  smtp server? I saw this at uninbox, they are also using postalserver, and you just have to create an mx record to their postal instance, and you can then use your own domain. Other services like skiff are working also like this. And what  do you mean with „CNAME them  to the records…“?

**Reply by i-am-henri** [reply]
The email provider https://uninbox.com has the postal web dashboard  on https://zero.e.uninbox.com.
![IMG_0650](https://github.com/user-attachments/assets/0bb29b7f-73ad-41d9-9f26-353f6c4998c5)
![IMG_0652](https://github.com/user-attachments/assets/ce672d69-46e5-4c70-9ca9-5c0a93617236)
On one of their domains https://unin.me, they are just pointing with an mx record to this server:
![IMG_0648](https://github.com/user-attachments/assets/a964fc31-db45-44b5-b340-c139df94811e)
![IMG_0649](https://github.com/user-attachments/assets/5506821e-40fd-49ac-9c7e-af07914ae14d)
My question was to archive the same, so I don‘t have to configure each domain on the web dashboard.
@willpower232

**Reply by willpower232** [reply]
If you follow the instructions on the docs, you end up with a set of domains pointing at your servers IP address https://docs.postalserver.io/getting-started/dns-configuration

You can then use these in MX and CNAME form as told to you by the Postal web UI

![image](https://github.com/user-attachments/assets/cede5d5c-4044-41ca-ab82-1cf63502f72e)

I'm not sure where you are getting confused?

**Reply by i-am-henri** [reply]
I thought I had set everything up correctly. I created the A record to point the mx records to, and the postal web dashboard also returns no errors on this domain. @willpower232 
![IMG_0655](https://github.com/user-attachments/assets/25ed5b5f-1bda-4007-b629-3eb0f685a761)
![IMG_0656](https://github.com/user-attachments/assets/52783b51-904b-454a-bccb-3dcf7c628756)

**Reply by willpower232** [reply]
Ah I see, so you should be good to add more domains and follow the instructions.

Referring to the original issue, have you checked mxtoolbox.com to see if you have been blocked be anyone?

**Reply by i-am-henri** [reply]
@willpower232  But it's not possible to just link to the domain? Or can I automate it (the process with adding every domain to the dashboard) with an api?

**Reply by willpower232** [reply]
Postal has to know about the domain so it can authorise the sending of the message (rather than just accepting anything) and also provides a different DKIM record for each domain.

There isn't (currently) an API for domains, and even if you bulk inserted them some other way, you'd still want to know what DKIM record Postal had generated for each one.
